glean-core/src/database/mod.rs

```diff
@@ -173,9 +179,13 @@ use crate::Result;
 /// before data is flushed to disk.
 ///
 /// Only considered if `delay_ping_lifetime_io` is set to `true`.
-#[cfg(target_os = "android")]
+#[cfg(any(target_os = "android", feature = "auto_flush"))]
 const PING_LIFETIME_THRESHOLD: usize = 1000;
 
+// Save atleast every 2 seconds.
+#[cfg(any(target_os = "android", feature = "auto_flush"))]
+const PING_LIFETIME_MAX_TIME: Duration = Duration::from_millis(2000);
```

Comment:
I chose this value arbitrarily, same as the 1000 before. Should we make this a config option by now?

Reply:
What kind of config? At-init runtime config? Or compile-time?

Reply:
at-init runtime means we could configure that from the fenix side without a glean release.
(Rust is not good at compile-time configuration options so I'd rather not come up with a method to do that right now)